tests/pauli_rotations/circuit_test.py

```diff
@@ -57,6 +65,74 @@ def test_count_rotations_by(circuit, fraction, expected):
     assert circuit.count_rotations_by(fraction) == expected
 
 
-# def test_remove_y_operators_from_circuit(output, expected):
-#     raise NotImplementedError
-#     # assert output == expected
+@pytest.mark.parametrize(
+    "rotation1, rotation2",
+    [
+        (([X], Fraction(1, 4)), ([I], Fraction(1, 4))),
+        (([I, X], Fraction(1, 4)), ([Y, X], Fraction(-1, 8))),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Y], Fraction(1, 8))),
+        (([I, Z, I, X], Fraction(-1, 4)), ([Z, X, I, Z], Fraction(-1, 8))),
+    ],
+)
+def test_commute_rotation_to_rotation_commuting(rotation1, rotation2):
+    test_circuit = PauliOpCircuit(len(rotation1[0]))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation1))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation2))
+    test_circuit.commute_pi_over_four_rotation(0)
+
+    assert test_circuit.ops[0].ops_list == rotation2[0]
+    assert test_circuit.ops[0].rotation_amount == rotation2[1]
+    assert test_circuit.ops[1].ops_list == rotation1[0]
+    assert test_circuit.ops[1].rotation_amount == rotation1[1]
+
+
+@pytest.mark.parametrize(
+    "rotation1, rotation2, resultant_rotation",
+    [
+        (([X], Fraction(1, 4)), ([Z], Fraction(1, 4)), ([Y], Fraction(1, 4))),
+        (([Y], Fraction(1, 4)), ([X], Fraction(-1, 8)), ([Z], Fraction(-1, 8))),
+        (([Z, X], Fraction(1, 4)), ([Y, X], Fraction(1, 8)), ([X, I], Fraction(1, 8))),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Z], Fraction(-1, 8)), ([Z, Y, X], Fraction(-1, 8))),
+        (
+            ([X, Y, X, Z, Z], Fraction(1, 4)),
+            ([Z, Y, I, X, Y], Fraction(1, 8)),
+            ([Y, I, X, Y, X], Fraction(1, 8)),
+        ),
```

Comment:
For this and other anti-commuting cases, reviews are needed on the sign of the resultant rotation. According to https://github.com/latticesurgery-com/lattice-surgery-compiler/issues/28#issuecomment-786611605, it should be -pi/8 for  line 99?

Reply:
It does look like the sign should be changing

Reply:
Another thing to fix before merging this

Reply:
I think the mistake is here:
https://github.com/latticesurgery-com/lattice-surgery-compiler/blob/d0c60f3299257837b963958e24955177eb5ddcf0/src/lsqecc/pauli_rotations/circuit.py#L233-L235

Reply:
but I still need to go through it properly

Reply:
The resultant rotation should have the fraction as +1/8. That is the correct answer.

Reply:
There should also be a case to test where the sign of the `resulltant_rotation` gets flipped.. All the examples here retain the sign of `rotation_2`. For example, in line 95 exchanging one Pauli Opertor between the rotations should flip the sign of the `resultant_rotation`. Like below
```
        (([X, Z, Z], Fraction(1, 4)), ([Y, X, Y], Fraction(-1, 8)), ([Z, Y, X], Fraction(1, 8))),
```

Reply:
What about the calculations in 28, they make it look like the sign should be negative?

Reply:
Ok I see the mistake
![Screenshot_20211201_121720-1](https://user-images.githubusercontent.com/36427091/144307988-75d28080-b37a-4ec5-8fcd-3bb609b4317f.png)

Reply:
so the problem is with the example not the code